dotcom-rendering/src/web/layouts/NewsletterSignupLayout.tsx

```diff
@@ -101,6 +115,77 @@ const leftColWrapperStyle = css`
 	}
 `;
 
+const previewButtonWrapperStyle = css`
+	padding: ${space[2]}px 0;
+	${from.desktop} {
+		display: none;
+	}
+`;
+
+const mainGraphicWrapperStyle = css`
+	border-radius: ${space[2]}px;
+	overflow: hidden;
+	margin: ${space[4]}px 0;
+
+	& > figcaption {
+		display: flex;
+
+		${until.desktop} {
+			display: none;
+		}
+
+		background-color: ${brandAlt[400]};
+		align-items: center;
+		padding: ${space[1]}px;
+		${textSans.medium({ fontWeight: 'bold', lineHeight: 'tight' })}
+
+		svg {
+			margin-right: ${space[1]}px;
+			flex-shrink: 0;
+		}
+	}
+`;
+
+interface NewsletterDetailProps {
+	text: string;
+	showUntil?: Breakpoint;
+}
+const NewsletterDetail = ({ text, showUntil }: NewsletterDetailProps) => {
```

Comment:
It's classically hard to know when this is a good idea or not - and I'm not really on top of this code enough to make the call myself - but you could consider if this component need breaking out into it's own file?

Pros: You can create a story for it, it's easier to expand later
Cons: It separates this logic away from where it's being used and hides it.

My instinct would be to break it out. It's nice if the layout file has a little css in it as possible and is only positioning components on the page but I'm happy to leave the call with you.